odpc.client/src/composables/use-form-validator.ts

```diff
@@ -0,0 +1,100 @@
+import { ref, onUnmounted, type ComputedRef, watchEffect } from "vue";
+
+export const useFormValidator = (formRef: ComputedRef<HTMLFormElement | undefined>) => {
+  const isValid = ref(false);
+
+  const validateForm = () => {
+    if (!formRef.value) {
+      isValid.value = false;
+      return;
+    }
+
+    isValid.value = validateCheckboxGroups() && formRef.value.checkValidity();
+
+    if (!isValid.value) {
+      focusInvalidInput(formRef.value);
+    }
+  };
+
+  const validateCheckboxGroups = () => {
+    const checkboxGroups = getCheckboxGroups();
+
+    if (checkboxGroups.length === 0) return true;
+
+    return Array.from(checkboxGroups).every((group) => {
+      group.classList.remove("invalid");
+
+      const checkboxes = getCheckboxesFromGroup(group);
+
+      if (!isAnyChecked(checkboxes)) {
+        group.classList.add("invalid");
+        group.open = true;
+
+        checkboxes[0]?.focus();
+
+        return false;
+      }
+
+      return true;
+    });
+  };
+
+  // Input type text
+  const focusInvalidInput = (form: HTMLFormElement) => {
+    const invalidInputs = form.querySelectorAll(
+      "input[type='text']:invalid"
+    ) as NodeListOf<HTMLInputElement>;
+
+    invalidInputs.forEach((input) => {
+      input.classList.add("invalid");
```

Comment:
kunnen we hier iets met de `:user-invalid` selector in plaats van een class? die triggert ook als je langs een input tabt

Reply:
Ja, dank voor tip, toegepast.